theater/frontline.py

```diff
@@ -1,47 +1,2 @@
-"""Battlefield front lines."""
-from typing import Tuple
-
-from dcs.mapping import Point
-from . import ControlPoint, MissionTarget
-
-# TODO: Dedup by moving everything to using this class.
-FRONTLINE_MIN_CP_DISTANCE = 5000
-
-
-def compute_position(control_point_a: ControlPoint,
-                     control_point_b: ControlPoint) -> Point:
-    a = control_point_a.position
-    b = control_point_b.position
-    attack_heading = a.heading_between_point(b)
-    attack_distance = a.distance_to_point(b)
-    middle_point = a.point_from_heading(attack_heading, attack_distance / 2)
-
-    strength_delta = float(control_point_a.base.strength -
-                           control_point_b.base.strength)
-    position = middle_point.point_from_heading(attack_heading,
-                                               strength_delta *
-                                               attack_distance / 2 -
-                                               FRONTLINE_MIN_CP_DISTANCE)
-    return position
-
-
-class FrontLine(MissionTarget):
-    """Defines a front line location between two control points.
-
-    Front lines are the area where ground combat happens.
-    """
-
-    def __init__(self, control_point_a: ControlPoint,
-                 control_point_b: ControlPoint) -> None:
-        super().__init__(f"Front line {control_point_a}/{control_point_b}",
-                         compute_position(control_point_a, control_point_b))
-        self.control_point_a = control_point_a
-        self.control_point_b = control_point_b
-
-    @property
-    def control_points(self) -> Tuple[ControlPoint, ControlPoint]:
-        """Returns a tuple of the two control points."""
-        return self.control_point_a, self.control_point_b
-
-    def is_friendly(self, to_player: bool) -> bool:
-        return False
+"""Only here to keep compatibility for save games generated in version 2.2.0"""
```

Comment:
Could it not be left here?

Reply:
Putting Discord conversation here in case we decide to revisit/refactor:

> I moved Frontline in with ConflictTheater because they must be tightly coupled to store the frontline data in the ConflictTheater instance data.  This requires circular imports when in separate modules.
> 
> The other way I tried was storing the frontline data as ConflictTheater class variable, but that isn't pickled so doesn't maintain state across saves if you were to change the source .miz/json.
> 
> Basically, if we want the frontline shape to be saved with the pickled data, they must be tightly coupled, and I'd prefer a larger module over circular imports.
> Alternatively, it might be possible to move all the references to FrontLine out of `ConflictTheater', but then we need to change a bunch of other calls.
> ColonelPanicToday at 8:51 PM
> Good enough reason for me
> Just don't normally see refactoring to fewer files, so struck me as odd :slight_smile:
> Go ahead and ignore that comment then